### PR TITLE
Feature/4779 finalize flow

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Finalized the flow and error handling.

## Link to Jira
[4779](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4779)

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift [hunk 1]
```diff
@@ -13,7 +13,8 @@ final class ExposureDetectionCoordinator {
 	private let homeState: HomeState
 	private let exposureManager: ExposureManager
 	private let otpService: OTPServiceProviding
-	
+	private let surveyURLProvider: SurveyURLProvidable
+
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift:17]: maybe change SurveyURLProvidable -> SurveyURLProviding to be consistent with the namings across the project

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/__tests__/Survey/SurveyURLServiceTests.swift [hunk 2]
```diff
@@ -7,13 +7,13 @@ import XCTest
 
 class SurveyURLServiceTests: XCTestCase {
 
-	func test_WHEN_getURLIsCalled_THEN_AnURLIstReturned() throws {
+	func test_WHEN_getURLIsCalled_THEN_AnURLIstReturned() {
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/__tests__/Survey/SurveyURLServiceTests.swift:10]:
```suggestion
	func test_WHEN_getURLIsCalled_THEN_aURLIstReturned() {
```

**Review by 30mar — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift:17]: done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/__tests__/Survey/SurveyURLServiceTests.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/__tests__/Survey/SurveyURLServiceTests.swift]: done

**Review by 30mar — APPROVED** [review]

**Review by janetback — COMMENTED** [review]
UA reviewed

**Pull request merged**